src/riak_core_vnode_worker.erl

```diff
@@ -37,12 +38,11 @@
         modstate :: any()
     }).
 
--spec behaviour_info(atom()) -> 'undefined' | [{atom(), arity()}].
-behaviour_info(callbacks) ->
-    [{init_worker,3},
-     {handle_work,3}];
-behaviour_info(_Other) ->
-    undefined.
+-callback init_worker(partition(), Args :: term(), Props :: [{atom(), term()}]) ->
```

Comment:
I'm not sure if it's really important, since every implementation of `init_worker` that we have ignores the `Props` argument 😆 . I guess it depends on whether we want to encourage people to use `proplists` or not, and I'm not a huge fan of the `proplists` module, so my vote would be to leave it as is...but admittedly that's fairly arbitrary ;). If you feel strongly about it, I can change it. Or we can also always leave it for now, and then widen the spec in the future if we decide we want to make it more permissive.